server/src/main/java/org/elasticsearch/action/admin/cluster/bootstrap/TransportGetDiscoveredNodesAction.java

```diff
@@ -124,4 +125,16 @@ public String toString() {
             });
         }
     }
+
+    private static boolean checkWaitRequirements(GetDiscoveredNodesRequest request, Set<DiscoveryNode> nodes) {
+        Set<String> requiredNodes = new HashSet<>(request.getRequiredNodes());
+        for (final DiscoveryNode node : nodes) {
```

Comment:
We should return an exception if any requirement matches >1 node.

I also wonder if we should return an exception if we specified a set of nodes and found nodes that didn't match this set. This statement is a note to myself to check this, no action is required yet.